From 941aa8a662107a66188f9633401afd4239f5bce3 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 5 Jul 2017 12:20:33 +0800
Subject: [PATCH 2105/5242] MLK-15325: pxp-v3: Modify pxp pitch parameter and
 csc coefficient setting.

commit  3eff3a6d64362e800b48257766e58beb158a05c8 from
https://source.codeaurora.org/external/imx/linux-imx.git

Because the caller of pxp-v3 does not set the stride parameter,
this will cause pitch parameter to be zero and pxp can't work.

Correct the csc1 coefficient when use pxp convert YUV to RGB format.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index e8b5018..b98be67 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -2647,7 +2647,7 @@ static int pxp_2d_task_config(struct pxp_pixmap *input,
 			pxp_fetch_config(input, position);
 			break;
 		case PXP_2D_CSC1:
-			pxp_csc1_config(input, false);
+			pxp_csc1_config(input, true);
 			break;
 		case PXP_2D_ROTATION1:
 			pxp_rotation1_config(input);
@@ -3217,8 +3217,8 @@ static int convert_param_to_pixmap(struct pxp_pixmap *pixmap,
 	pixmap->height = param->height;
 	pixmap->format = param->pixel_fmt;
 	pixmap->paddr  = param->paddr;
-	pixmap->pitch  = param->stride;
 	pixmap->bpp    = get_bpp_from_fmt(pixmap->format);
+	pixmap->pitch  = param->width * pixmap->bpp >> 3;
 
 	pixmap->crop.x = param->left;
 	pixmap->crop.y = param->top;
-- 
1.7.9.5

